tests/components/template/test_switch.py

```diff
@@ -563,3 +609,38 @@ async def test_invalid_availability_template_keeps_component_available(hass, cap
 
     assert hass.states.get("switch.test_template_switch").state != STATE_UNAVAILABLE
     assert ("UndefinedError: 'x' is undefined") in caplog.text
+
+    def test_restore_state(self):
```

Comment:
Please make this test a standalone async pytest test coroutine function instead. It needs to be that for the storage to be cleaned up properly.

See eg:
https://github.com/home-assistant/core/blob/2af984917e7cf3ee649ed3a531a872e7aaeba713/tests/components/input_boolean/test_init.py#L123-L145

Reply:
I tried that. It was hard work to make this test pass actually, because the `hass.states.get` would return `None` all the time. Until I added the `wait_till_done`. Now, I'm back with the dirty files left behind...